docs/en/flash.md

```diff
@@ -48,24 +48,32 @@ Otherwise, if you built your own firmware from source code:
 
 Also, in some special circumstances, you may need to flash `blank.bin` or `esp_init_data_default.bin` to various addresses on the flash (depending on flash size and type), see [below](#upgrading-from-sdk-09x-firmware).
 
-If upgrading from [SPIFFS](https://github.com/pellepl/spiffs) version 0.3.2 to 0.3.3 or later, or after flashing any new firmware (particularly one with a much different size), you may need to run [`file.format()`](modules/file.md#fileformat) to re-format your flash filesystem. You will know if you need to do this if your flash files disappeared, or if they exist but seem empty, or if data cannot be written to new files.
-
-## Upgrading from SDK 0.9.x Firmware
+## Upgrading Firmware
 
 !!! important
 
     It goes without saying that you shouldn't expect your NodeMCU 0.9.x Lua scripts to work error-free on a more recent firmware. Most notably Espressif changed the `socket:send` operation to be asynchronous i.e. non-blocking. See [API documentation](modules/net.md#netsocketsend) for details.
 
-If you flash a recent NodeMCU firmware for the first time, it's advisable that you get all accompanying files right. A typical case that often fails is when a module is upgraded from a 0.9.x firmware to the latest version built from the [NodeMCU build service](http://nodemcu-build.com). It might look like the brand new firmware is broken, but the reason for the missing Lua prompt is related to the big jump in SDK versions: Espressif changes the `esp_init_data_default.bin` for their devices along the way with the SDK. So things break when a NodeMCU firmware with a certain SDK is flashed to a module which contains init data from a previous SDK.
+Espressif changes the init data block (`esp_init_data_default.bin`) for their devices along the way with the SDK. So things break when a NodeMCU firmware with a certain SDK is flashed to a module which contains init data from a different SDK. Hence, this section applies to upgrading NodeMCU firmware just as well as *downgrading* firmware.
+
+A typical case that often fails is when a module is upgraded from a 0.9.x firmware to a recent version. It might look like the new firmware is broken, but the reason for the missing Lua prompt is related to the big jump in SDK versions.
+
+If there is no init data block found during SDK startup, the SDK will install one itself. If there is a previous (potentially too old init block), the SDK *probably* doesn't do anything with it but there is no documentation from Espressif on this topic.
+
+Hence, there are two strategies to update the SDK init data:
+
+- Erase flash completely. This will also erase the (Lua) files you uploaded to the device! The SDK will install the init data block during startup.
```

Comment:
This isn't the case for either my esp-07(512k) or my esp-12(4M), after I erase the flash with `esptool.py ... erase_flash` then upload `0x00000.bin` and `0x10000.bin` with `esptool.py ... write_flash`, I get the error `rf_cal[0] !=0x05,is 0xFF` then it just spits out garbage after I pull the reset pin low.

In order to get the esp-07 to boot up I had to flash `esp_init_data_default.bin` to `0x7c000` and it booted up, simple enough.
The esp-12 was not so simple, I flashed `esp_init_data_default.bin` to `0x3fc000` and still got garbage, so I erased again then flashed it another time and got the same result. I spent the whole day messing with it and got nowhere.
Then I got the bright idea to try flashing `esp_init_data_default.bin` to `0x7c000` and `0x3fc000` then it finally booted up.

I'm not sure if this is a problem with the SDK or NodeMCU.